llvm/lib/Transforms/Vectorize/LoopVectorizationLegality.cpp

```diff
@@ -1442,6 +1487,126 @@ bool LoopVectorizationLegality::canVectorizeLoopNestCFG(
   return Result;
 }
 
+bool LoopVectorizationLegality::isVectorizableEarlyExitLoop() {
+  // At least one of the exiting blocks must be the latch.
```

Comment:
Below just checks if there's a latch, not if it is exiting?

Reply:
Hmm, I should move the comment lower down to the code where it's checked:

```
  if (isa<SCEVCouldNotCompute>(
          PSE.getSE()->getPredicatedExitCount(TheLoop, LatchBB, &Predicates))) {
    reportVectorizationFailure(
        "Cannot determine exact exit count for latch block",
        "Cannot vectorize early exit loop",
        "UnknownLatchExitCountEarlyExitLoop", ORE, TheLoop);
    return false;
  }
```

There is a test for this - see `@early_exit_infinite_loop`

Reply:
Done